### PR TITLE
fix: [CI-5606]: Fetch concurrent executions for CI build types

### PR DESCRIPTION
**Pull request by smjt-h**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40020)

### CODE REVIEW

**Review by aman-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40020)* status: 5 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`332-ci-manager/service/src/main/java/io/harness/ci/execution/execution/CIAccountBuildMetadata.java` line 31 at r1](https://reviewable.io/reviews/harness/harness-core/40020#-NGoguAl4YJu--p3HlZN:-NGoguAl4YJu--p3HlZO:b-jagkjx) ([raw file](https://github.com/harness/harness-core/blob/69dd7b90a947b01209759b27e921228cf4e42f30/332-ci-manager/service/src/main/java/io/harness/ci/execution/execution/CIAccountBuildMetadata.java#L31)):*
> ```Java
> @TypeAlias("ciAccountBuildMetadata")
> @HarnessEntity(exportable = true)
> public class CIAccountBuildMetadata {
> ```

can we rename it to CIExecutionMetadata?
___
*[`332-ci-manager/service/src/main/java/io/harness/ci/execution/execution/CIAccountBuildMetadata.java` line 34 at r1](https://reviewable.io/reviews/harness/harness-core/40020#-NGoho3M93qrBEtWpuyC:-NGoho3M93qrBEtWpuyD:b-ymzuc5) ([raw file](https://github.com/harness/harness-core/blob/69dd7b90a947b01209759b27e921228cf4e42f30/332-ci-manager/service/src/main/java/io/harness/ci/execution/execution/CIAccountBuildMetadata.java#L34)):*
> ```Java
>   @Wither @Id @org.mongodb.morphia.annotations.Id String uuid;
>   String accountId;
>   OSType buildType;
> ```

Can we not hardcode the key as OSType and generate it in code and save it here? It could be of type K8s as well.

**Review by aman-harness — COMMENTED** [review]
Please make sure that the indexes are created. 

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40020#-:-NGokJGg3zlEbER8_ew2:b-74ih3k)* status: 5 file(s) to review. 1 change requests. 0 of 1 approvals obtained.

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/execution/QueueExecutionUtils.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import io.harness.beans.yaml.extended.infrastrucutre.VmInfraYaml;
+import io.harness.beans.yaml.extended.infrastrucutre.VmPoolYaml;
+import io.harness.plancreator.steps.common.StageElementParameters;
+import io.harness.pms.sdk.core.events.OrchestrationEvent;
+import io.harness.repositories.CIAccountBuildRepository;
+
+import com.google.inject.Inject;
+
+public class QueueExecutionUtils {
+  @Inject private CIAccountBuildRepository ciAccountBuildRepository;
```
Comment (aman-harness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/execution/QueueExecutionUtils.java:20]: CIExecutionRepository Pls rename

Comment (smjt-h) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/execution/QueueExecutionUtils.java]: done

**Review by aman-harness — APPROVED** [review]

**Review by aman-harness — APPROVED** [review]
> Reviewed 2 of 5 files at r2, 5 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40020)* status: :shipit: complete! ✓

**Pull request merged**